### PR TITLE
ci: :construction_worker: Add ABI check to ros_indiustrial ci

### PR DESCRIPTION
**Pull request by jaron-l**
refs: #583

### CODE REVIEW

File: .github/workflows/binary.yml [hunk 1]
```diff
@@ -7,17 +7,20 @@ on:
...
 jobs:
   ci_binary:
     name: Galactic binary job
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        env:
-          - {ROS_DISTRO: galactic, ROS_REPO: main}
-          - {ROS_DISTRO: galactic, ROS_REPO: testing}
+          ROS_REPO: [main, testing]
```
Comment (bmagyar) [inline .github/workflows/binary.yml:20]: can we have this on a separate workflow called "ABI Compatibility" or "ABI Check" and have the tests disabled?

File: .github/workflows/binary.yml [hunk 2]
```diff
@@ -7,17 +7,20 @@ on:
   schedule:
     # Run every morning to detect flakiness and broken dependencies
     - cron: '17 8 * * *'
+env:
+  ROS_DISTRO: galactic
+  BRANCH: master
```
Comment (bmagyar) [inline .github/workflows/binary.yml:12]: Where ABI stability is desired are:

foxy branch for the foxy distro
galactic branch for the galactic distro

the master branch targets the rolling distro where we have absolutely no intentions in keeping ABI stability

**Review by bmagyar — CHANGES_REQUESTED** [review]

File: .github/workflows/binary.yml (hunk 2 above)
Comment (bmagyar) [reply .github/workflows/binary.yml:12]: *actually* we need "current branch" to whatever the target is

File: .github/workflows/binary.yml [hunk 3]
```diff
@@ -7,17 +7,20 @@ on:
...
-          - {ROS_DISTRO: galactic, ROS_REPO: testing}
+          ROS_REPO: [main, testing]
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
       - uses: ros-industrial/industrial_ci@master
-        env: ${{matrix.env}}
+        env:
+          ROS_REPO: ${{matrix.ROS_REPO}}
+          ABICHECK_URL: github:ros-controls/ros2_control#${{ env.BRANCH }}
```
Comment (destogl) [inline .github/workflows/binary.yml:26]: I propose use of `github.base_ref` (see [here](https://docs.github.com/en/actions/learn-github-actions/contexts)). 

Also we only need this to run on pull request.

File: .github/workflows/binary.yml (hunk 2 above)
Comment (jaron-l) [reply .github/workflows/binary.yml:12]: Have we already branched for galactic?

I was thinking that I would just submit a new pull request for each branch to specify the target branch once this one has been approved.

Comment (bmagyar) [reply .github/workflows/binary.yml:12]: yep, branched for galactic over a month ago

**Review by destogl — APPROVED** [review]

**Review by bmagyar — APPROVED** [review]

**Comment by bmagyar** [comment]
@mergifyio backport to foxy please

**Pull request merged**